spec/services/case_remove_sar_deadline_extension_service_spec.rb

```diff
@@ -0,0 +1,81 @@
+require "rails_helper"
+
+describe CaseRemoveSARDeadlineExtensionService do
+  let(:team_dacu)           { find_or_create :team_disclosure_bmt }
+  let(:manager)             { find_or_create :disclosure_bmt_user }
+  let!(:sar_case)           { create(:extended_deadline_sar) }
+  let!(:initial_deadline)   { sar_case.initial_deadline }
+  let!(:max_extension)      { Settings.sar_extension_limit.to_i }
+
+  before do
+    allow(sar_case.state_machine).to receive(:remove_sar_deadline_extension!)
+  end
+
+  describe '#initialize' do
+    subject { new_service(manager, sar_case) }
+
+    it { expect(sar_case.external_deadline).not_to eq initial_deadline }
+    it { expect(subject.result).to eq :incomplete }
+  end
+
+  describe '#call' do
+    context 'with expected params' do
+      subject! { new_service(manager, sar_case).call }
+
+      it { is_expected.to eq :ok }
+      it 'creates new SAR extension removal transition' do
+        expect(sar_case.state_machine)
+          .to have_received(:remove_sar_deadline_extension!)
+          .with(
+            acting_user: manager,
+            acting_team: team_dacu
+          )
+      end
+
+      it 'resets SAR deadline date' do
+        expect(sar_case.external_deadline).to eq initial_deadline
+      end
+    end
+
+    context 'after initial deadline' do
+      it 'allows retrospective removal' do
+        expect(sar_case.external_deadline).to be > sar_case.initial_deadline
+        expect(sar_case.deadline_extended?).to be true
+
+        Timecop.travel(sar_case.external_deadline + 100.days) do
+          result = new_service(manager, sar_case).call
+
+          expect(result).to eq :ok
+          expect(sar_case.external_deadline).to eq initial_deadline
+          expect(sar_case.deadline_extended?).to be false
+        end
+      end
+    end
+
+    context 'exception' do
```

Comment:
what exception is this?

Reply:
Renamed with more descriptive context name